File: epi_judge_java/epi/Fibonacci.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class Fibonacci {

    @EpiTest(testDataFile = "fibonacci.tsv")
    public static int fibonacci(int n) {
        if (n < 2) return n;
        int a = 0;
        int b = 1;

        for (int i = 2; i <= n; i++) {
            int temp = a;
            a = b;
            b += temp;
        }

        return b;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "Fibonacci.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_cpp/fibonacci.cc
#include "test_framework/generic_test.h"

int Fibonacci(int n) {
    if (n < 2) return n;
    int a = 0;
    int b = 1;
    for (int i = 1; i < n; i++) {
        int temp = a;
        a = b;
        b += temp;
    }
    return b;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"n"};
    return GenericTestMain(args, "fibonacci.cc", "fibonacci.tsv", &Fibonacci,
                           DefaultComparator{}, param_names);
}
